Source/Plugins/Core/com.equella.core/js/tsrc/mainui/App.tsx

```diff
@@ -0,0 +1,264 @@
+/*
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * The Apereo Foundation licenses this file to you under the Apache License,
+ * Version 2.0, (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ThemeProvider } from "@material-ui/core";
+import { oeqTheme } from "../theme";
+import * as React from "react";
+import { getCurrentUser, UserData } from "../api/currentuser";
+import { ErrorResponse } from "../api/errors";
+import {
+  LegacyContent,
+  LegacyContentProps,
+  PageContent,
+} from "../legacycontent/LegacyContent";
+import { Template, TemplateProps, TemplateUpdate } from "./Template";
+import { defaultNavMessage, NavAwayDialog } from "./PreventNavigation";
+import { shallowEqual } from "shallow-equal-object";
+import { OEQRoute, OEQRouteComponentProps, routes } from "./routes";
+import {
+  Prompt,
+  Redirect,
+  Route,
+  RouteComponentProps,
+  Switch,
+} from "react-router";
+import ErrorPage from "./ErrorPage";
+import { LegacyPage, templatePropsForLegacy } from "./LegacyPage";
+import { BrowserRouter } from "react-router-dom";
+import { LegacyForm } from "../legacycontent/LegacyForm";
+import HtmlParser from "react-html-parser";
+import SettingsPage from "../settings/SettingsPage";
+import { startHeartbeat } from "../util/heartbeat";
+
+const beforeunload = function (e: Event) {
+  e.returnValue = ("Are you sure?" as unknown) as boolean;
```

Comment:
Using the `BeforeUnloadEvent` should remove the need for casing string to boolean.

```suggestion
const beforeunload = function (e: BeforeUnloadEvent) {
  e.returnValue = "Are you sure?";
```